File: tests/Kivy_test/api_call.py
import os, json, requests
import matplotlib.pyplot as plt

base_url = r"https://power.larc.nasa.gov/api/temporal/daily/point?parameters=T2M,T2MDEW,T2MWET,TS,T2M_RANGE,T2M_MAX,T2M_MIN&community=RE&longitude={longitude}&latitude={latitude}&start=20150101&end=20150305&format=JSON"

class ApiCall():
    def __init__(self,lon = 32.929, lat=-95.770):
        self.lon = lon
        self.lat = lat
                
    def fetch_data(self):
        '''Makes a req to api returns dict'''
        api_req_url = base_url.format(longitude=self.lon, latitude=self.lat)
        response = requests.get(url=api_req_url, verify= True, timeout=45.00)
        content = json.loads(response.content.decode('utf-8'))
        
        return content

    def unpack_data(self):
        '''looks at the dict obtained from api call, extract temp data + dates'''
        raw_data = self.fetch_data()
        temp_data = list(raw_data['properties']['parameter']['T2M'].values())
        dates = list(raw_data['properties']['parameter']['T2M'].keys())
        return temp_data, dates
    
    

File: tests/pro-app/src/utils/APIcall.js
import React from 'react'
import axios from 'axios'

const lat = 43.7068;
const lon = -79.3985;
const test_url1 = 'https://power.larc.nasa.gov/api/temporal/daily/point?parameters=T2M&community=RE&longitude=-75.7097&latitude=45.3928&start=20210101&end=20210131&format=JSON'
const test_url2 = 'https://power.larc.nasa.gov/api/temporal/daily/point?parameters=T2M&community=RE&longitude=-123.1706&latitude=49.3199&start=20210101&end=20210131&format=JSON'
const test_url3 ='https://power.larc.nasa.gov/api/temporal/daily/point?parameters=T2M&community=RE&longitude=-123.4343&latitude=48.4239&start=20210101&end=20210331&format=JSON'
formated_url = (lon, lat) => `https://power.larc.nasa.gov/api/temporal/daily/point?parameters=T2M,T2MDEW,T2MWET,TS,T2M_RANGE,T2M_MAX,T2M_MIN&community=RE&longitude=${lon}&latitude=${lat}&start=20150101&end=20150305&format=JSON`;
const url = formated_url(lon, lat)

const updateData = (response) => {
    // Takes the response of the api call, extracts, splits
    // then updates data states for plotting
    const T2M = response.data.properties.parameter.T2M
    const dates = Object.keys(T2M)
    const temps = Object.values(T2M)
    setLabel(dates), setData(temps)
  }

  const apiCall = () => {
    axios.get(test_url2)
      .then(response => {
      updateData(response);
    }, error => {
      console.log(error)
    });
  }